index.bs

```diff
@@ -771,8 +771,9 @@ spec:css-syntax-3;
 
     2.  Assert: |settings| is a [=secure context=].
 
-    3.  If <code>|options|.{{CredentialRequestOptions/signal}}</code>'s [=AbortSignal/aborted flag=]
-        is set, then return [=a promise rejected with=] an "{{AbortError}}" {{DOMException}}.
+    3.  If <code>|options|.{{CredentialRequestOptions/signal}}</code> is [=AbortSignal/aborted=],
+        then return [=a promise rejected with=]
+        <code>|options|.{{CredentialRequestOptions/signal}}</code>'s [=AbortSignal/abort reason=].
```

Comment:
This seems fine, but we are not also addressing the final requirement enumerated by DOM's "[3.3. Using AbortController and AbortSignal objects in APIs](https://dom.spec.whatwg.org/#abortcontroller-api-integration)" which is: 
> Use the [abort algorithms](https://dom.spec.whatwg.org/#abortsignal-abort-algorithms) mechanism to observe changes to the AbortSignal object and do so in a manner that does not lead to clashes with other observers.

I'm not sure in practice this requirement definitely applies to this credman spec.

If it _does_ apply, then following the [example in the latter DOM spec section](https://dom.spec.whatwg.org/#abortcontroller-api-integration) leads me to **_guess_** that if we add these abort steps to `options.signal`:
```
1. Set `abortedFlag` to true.   // cause the overall alg to stop
2. Reject p with signal’s abort reason. 
```
...and then also create `abortedFlag` in the main synchronous portion of our "Request a `Credential`" alg, and then check in a couple/few places in the in-parallel portion of the "Request a `Credential`" alg whether our `abortedFlag` is true, and if so then say "abort these steps" (we wouldn't need to do the promise rejection because the "abort steps" we added take care of it).
 
Lacking a more detailed example I am just sort of guessing about this. I've [queried the DOM folks](https://github.com/whatwg/dom/issues/1030#issuecomment-965902194) about it in any case.

Reply:
I worried about this, but I think it's covered by the use of `options.signal` in [[Create]] and  [[DiscoverFromExternalSource]], as suggested by https://w3c.github.io/webappsec-credential-management/#implementation-extension.

Reply:
Okay, so if we decide to handle the signal being aborted in [[Create]] and [[DiscoveredFromExternalSource]], I think the changes on this PR are enough.

I don't think we need an `abortedFlag`, we can instead check `if signal is aborted`.

Reply:
Agreed, given @domenic's [comment](https://github.com/w3c/webappsec-credential-management/pull/176#pullrequestreview-804196800) below.